corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js

```diff
@@ -916,15 +916,18 @@ hqDefine("cloudcare/js/form_entry/entries", function () {
     FileEntry.prototype.constructor = EntrySingleAnswer;
     FileEntry.prototype.onAnswerChange = function (newValue) {
         var self = this;
-        if (newValue !== constants.NO_ANSWER) {
+        if (self.answer() && self.answer().match(/([a-z0-9]{8}-)([a-z0-9]{4}-){3}([a-z0-9]{12}).[a-z]{3}/)) {
```

Comment:
maybe add a comment here since this is not self explanatory without further thought

Reply:
does that match all video + file extensions? Is that name manipulated in the backend to rename to ensure that? I' just thinking into I rename a file to "video.mov" would that cause an issue?

Reply:
the comment is a good idea. added in: https://github.com/dimagi/commcare-hq/pull/32634/commits/331d43f0e20f84130d107fd74729079e329ac34c

Reply:
it matches the pattern of the unique id assigned by formplayer, which will look like `a9073a76-4b38-4e74-a7af-736d24fc3260` + `.mp4` or `.png` or even `.JpG` mixed case, which was something QA found can happen.
@shubham1g5 I figured it made sense to account for strange cases like `.jPG` etc on this end, but would it also make sense to convert file extensions to lowercase in `saveFile` in formplayer?

Reply:
cool sounds good thanks

Reply:
Regex nitpicks: the `.` ought to be escaped as `\.`, and can't the extension potentially be more than 3 characters? (jpeg, mpeg)

Does the regex need to be quite so specific? Something like `/[\w-]{32,48}\.\w{3,4}/` would make this code not reliant on the specific GUID format formplayer is using.

Reply:
Yep `.` should be escaped, missed that, thanks. 
The extensions can be more than 3 chars. This expression still matches on those, but I hadn't considered longer extensions until I saw that QA was testing with them. 
I went back and forth on how strict to make the matching criteria. I ended up tailoring it to the specific GUID format that formplayer is using. What I would worry about is someone uploading a file with a name that would cause unwanted behavior. For instance if a user selected `quitelongnamethatwillbreakthings.jpeg` to upload, it would match that expression, `/[\w-]{32,48}\.\w{3,4}/`, and not get sent to formplayer and fail to be saved. 
If we were to keep a strict check, I'd change it to something like `(\w{8}-)(\w{4}-){3}(\w{12})\.\w{3,4}`, but I'm open to making it less strict if you think it's overly cautious to match that format so closely.

Reply:
Ah, I misunderstood the purpose of the regex. So it needs to match the GUID-based filename sent back by formplayer but not the original filename, which is likely something human-readable. Then, yeah, I get why you're being stricter, I agree with your reasoning.